tests/test_flask_util.py

```diff
@@ -416,6 +442,32 @@ def two():
             rv = c.get('/two')
             self.assertEqual(rv.status_code, httplib.OK)
 
+    def test_incremental_auth_exchange(self):
+        self._create_incremental_auth_app()
+
+        with Http2Mock():
+            with self.app.test_client() as c:
+                with self.app.test_request_context():
```

Comment:
These ten lines of code also look similar but are in fact slightly different. Worth capturing in the same helper function?